lsp/src/parsers/json_schema/errors.rs

```diff
@@ -0,0 +1,84 @@
+pub fn additional_items_error(processed_items: usize, node_items: usize) -> String {
```

Comment:
might want look at [thiserror](https://docs.rs/thiserror/latest/thiserror/) which helps to generate proper error types.

Reply:
The purpose of these error functions aren't to help create rust errors, it's for the messages used when creating LSP diagnostics